project_event/models/project_task.py

```diff
@@ -215,6 +217,29 @@ def create_new_activity(self, vals):
             .next_by_code('project.task.activity')
         return super(Task, self).create(vals)
 
+    @api.multi
+    def write(self, vals):
+        if self.activity_task_type == 'activity':
+            self.write_task(vals)
+            self.write_activity(vals)
+        elif self.activity_task_type == 'task':
+            return super(Task, self).write(vals)
+
+    @api.multi
+    def write_activity(self, vals):
+        return super(Task, self).write(vals)
+
+    @api.multi
+    def write_task(self, vals):
+        main_task = self.get_main_task()
+        return main_task.write(vals)
+
+    def get_main_task(self):
+        return self.env['project.task'].search([
+            ('parent_id', '=', self.id),
+            ('name', '=', self.name)]
```

Comment:
what if we change the name of the activity or the main task? better to distinguish the main task by a boolean field (is_main_task)

Reply:
if we change the room id in main task by equipment and then we change the room id of the parent activity we will have a task with 2 resources (room and equipment) we must only have one resource per task

Reply:
@rimbendhaou :+1:  I have added the field and change accordingly